indexer/src/main/scala/com/expedia/www/haystack/trace/indexer/writers/es/ElasticSearchWriter.scala

```diff
@@ -118,7 +120,8 @@ class ElasticSearchWriter(esConfig: ElasticSearchConfiguration, indexConf: White
     var isSemaphoreAcquired = false
 
     try {
-      addIndexOperation(traceId, packedSpanBuffer.protoObj, ElasticSearchWriterUtils.indexName(esConfig.indexNamePrefix, esConfig.indexHourBucket), isLastSpanBuffer) match {
+      val eventTimeInMicros = packedSpanBuffer.protoObj.getChildSpansList.asScala.map(_.getStartTime).fold(Long.MaxValue)(Math.min)
```

Comment:
If we use the starttime of the head of the spanList, it may not impact the functionality, and we can avoid iterating over all the spans.

Reply:
done